### PR TITLE
chore(coderd/database/dbpurge): replace usage of time.* with quartz

### PR DESCRIPTION
**Pull request by johnstcn**
Related to https://github.com/coder/coder/issues/10576

This PR introduces quartz to `coderd/database/dbpurge` and updates the following unit tests to make use of Quartz's functionality:
- `TestPurge`
- `TestDeleteOldWorkspaceAgentLogs`

Additionally, updates `DeleteOldWorkspaceAgentLogs` to replace the hard-coded interval with a parameter passed into the query. This aids in testing and brings us a step towards allowing operators to configure the cutoff interval for workspace agent logs.

### CODE REVIEW

File: coderd/database/dbpurge/dbpurge_test.go [hunk 1]
```diff
@@ -177,57 +195,75 @@ func TestDeleteOldWorkspaceAgentLogs(t *testing.T) {
...
 		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitShort)
 		defer cancel()
+		clk := quartz.NewMock(t)
+		clk.Set(now).MustWait(ctx)
 
-		// given
-		agent := mustCreateAgentWithLogs(ctx, t, db, user, org, tmpl, tv, now.Add(-8*24*time.Hour), t.Name())
+		// After dbpurge completes, the ticker is reset. Trap this call.
+		trapReset := clk.Trap().TickerReset()
+		defer trapReset.Close()
```
Comment (mafredri) [inline coderd/database/dbpurge/dbpurge_test.go:203]: This seems shared between all tests, how about refactoring to a helper like below, or consider only testing it in one test and assuming it applies to the rest?

```go
	t.Helper()

	// After dbpurge completes, the ticker is reset. Trap this call.
	trapReset := clk.Trap().TickerReset()
	defer trapReset.Close()

	closer := do(clk)
	
	// Wait for the initial nanosecond tick.
	clk.Advance(time.Nanosecond).MustWait(ctx)

	trapReset.MustWait(ctx).Release() // Wait for ticker.Reset()

	return closer
```

Perhaps this example can even be taken further, just here as food for thought.

**Review by mafredri — APPROVED** [review]
LGTM

File: coderd/database/dbpurge/dbpurge_test.go (hunk 1 above)
Comment (johnstcn) [reply coderd/database/dbpurge/dbpurge_test.go:203]: I modify this test in the follow-up PR https://github.com/coder/coder/pull/14460 so this won't be an issue.

File: coderd/database/dbpurge/dbpurge_test.go [hunk 2]
```diff
@@ -29,26 +28,48 @@ import (
...
+	trapStop.Close()
+
+	require.NoError(t, purger.Close())
 }
 
 //nolint:paralleltest // It uses LockIDDBPurge.
 func TestDeleteOldWorkspaceAgentStats(t *testing.T) {
-	db, _ := dbtestutil.NewDB(t)
-	logger := slogtest.Make(t, &slogtest.Options{IgnoreErrors: true}).Leveled(slog.LevelDebug)
+	ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitShort)
```
Comment (dannykopping) [inline coderd/database/dbpurge/dbpurge_test.go:64]: Nit: `testutil.Context` is a nice convenience for this:

```go
func Context(t *testing.T, dur time.Duration) context.Context {
	ctx, cancel := context.WithTimeout(context.Background(), dur)
	t.Cleanup(cancel)
	return ctx
}
```

**Review by dannykopping — APPROVED** [review]
LGTM

File: coderd/database/dbpurge/dbpurge_test.go (hunk 2 above)
Comment (johnstcn) [reply coderd/database/dbpurge/dbpurge_test.go:64]: Will address in follow-up 👍

**Pull request merged**